crypto/ts/ts_rsp_sign.c

```diff
@@ -835,9 +859,132 @@ static int ESS_add_signing_cert(PKCS7_SIGNER_INFO *si, ESS_SIGNING_CERT *sc)
     return 0;
 }
 
-static ASN1_GENERALIZEDTIME
-*TS_RESP_set_genTime_with_precision(ASN1_GENERALIZEDTIME *asn1_time,
-                                    long sec, long usec, unsigned precision)
+static ESS_SIGNING_CERT_V2 *ESS_SIGNING_CERT_V2_new_init(const EVP_MD *hash_alg,
+                                                         X509 *signcert,
+                                                         STACK_OF(X509) *certs)
+{
+    ESS_CERT_ID_V2 *cid = NULL;
+    ESS_SIGNING_CERT_V2 *sc = NULL;
+    int i;
+
+    if ((sc = ESS_SIGNING_CERT_V2_new()) == NULL)
+        goto err;
+    if (sc->cert_ids == NULL &&
+            (sc->cert_ids = sk_ESS_CERT_ID_V2_new_null()) == NULL)
+        goto err;
+
+    if ((cid = ESS_CERT_ID_V2_new_init(hash_alg, signcert, 0)) == NULL
+        || !sk_ESS_CERT_ID_V2_push(sc->cert_ids, cid))
```

Comment:
_Small remark_: when the `sk_ESS_CERT_ID_V2_push` fails, you will have to delete yourself the unpushed item `cid`.
it is done below with this comment `/* Ownership is lost. */`